app/services/support_interface/locations_export.rb

```diff
@@ -0,0 +1,52 @@
+module SupportInterface
+  class LocationsExport
+    def data_for_export
+      application_choices.find_each.map do |application_choice|
+        application_form = application_choice.application_form
+
+        {
+          'Candidate id' => application_form.candidate_id,
+          'Support reference' => application_form.support_reference,
+          'Age' => return_age(application_form),
+          'Candidate’s postcode' => application_form.postcode,
+          'Provider’s postcode' => application_choice.provider.postcode,
+          'Site’s postcode' => application_choice.site.postcode,
+          'Provider type' => application_choice.provider.provider_type,
+          'Accrediting provider type' => application_choice.course.accredited_provider&.provider_type,
+          'Program type' => application_choice.course.program_type,
+          'Degree completed' => return_lastest_degree_award_year(application_form),
+          'Degree type' => return_lastest_degree_type(application_form),
+          'Status' => application_state(application_form),
+        }
+      end
+    end
+
+  private
+
+    def application_choices
+      ApplicationChoice
+      .joins(%i[application_form site provider course candidate])
+      .merge(Candidate.order(:id))
```

Comment:
I think you still need an `includes` to avoid the 1+n queries. Something like this:

```
ApplicationChoice
.includes(%i[application_form site provider course candidate])
.joins(:candidate)
.merge(Candidate.order(:id))
```